tests/test_type_errors.py

```diff
@@ -0,0 +1,188 @@
+"""
+This file contains a test allowing assertions to be made that an expected type error is in
+fact produced by the type-checker. I.e. that the type checker is not delivering a false
+negative.
+
+To use the test, add a comment of the following form to your test code:
+
+    # assert-type-error-pyright: 'No overloads for "execute_operation" match' await
+    nexus_client.execute_operation(  # type: ignore
+
+The `type: ignore` is only necessary if your test code is being type-checked.
+"""
+
+import itertools
+import json
+import os
+import platform
+import re
+import subprocess
+import tempfile
+from pathlib import Path
+
+import pytest
+
+
+def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
+    """Dynamically generate test cases for files with type error assertions."""
+    if metafunc.function.__name__ in [
+        "test_type_errors_pyright",
+        "test_type_errors_mypy",
+    ]:
+        tests_dir = Path(__file__).parent
+        files_with_assertions = []
+
+        for test_file in tests_dir.rglob("test_*.py"):
+            if test_file.name == "test_type_errors.py":
+                continue
+
+            if _has_type_error_assertions(test_file):
+                files_with_assertions.append(test_file)
+
+        metafunc.parametrize("test_file", files_with_assertions, ids=lambda f: f.name)
+
+
+@pytest.mark.skipif(platform.system() == "Windows", reason="TODO: broken on Windows")
```

Comment:
Can we open an issue to address this TODO so we don't lose track?

Reply:
Yes good call, done https://github.com/nexus-rpc/sdk-python/issues/19. There should be no technical obstacle there; I just failed to solve it within the time I had allotted to the task.